### ISSUE TITLE
On the memory consumption of DCCF training on ml20m

### ISSUE THREAD

**Issue by MeycL** [HKUDS/DCCF#2]
Hi, thank you very much for this wonderful work. I want to try to train on ml20m dataset with DCCF on RTX A5000 GPU (24GB), but it shows that there is not enough memory, is there any suggestion to solve it other than changing to a bigger GPU?

**Comment by Re-bin** [comment]
Hi!

Thanks for your interest in our work. Based on the information you provided, the problem may be caused by computing the gradient for the adaptive mask during the backpropagation process. If your dataset is dense, this can lead to insufficient GPU memory. 

In addition to upgrading to a larger GPU, you can try modifying line 85 of **model.py** to `return G_indices, G_values.detach()` to stop the computation of the gradient for the adaptive mask values, thus reducing memory usage. However, this modification may have implications for the performance of the model.

I hope the above answers can help with your confusion.

Best,
Xubin

**Comment by MeycL** [comment]
Thanks for your reply, I will try it.
